_drafts/wework-com-is-going-static.markdown

```diff
@@ -0,0 +1,113 @@
+
+
+### The static / dynamic hybrid approach
+
+One key requirement for us when evaluating different static site generators was the ability to hit an API endpoint and dynamically generate static pages based on the data returned. A real-world example of this is when we add new locations and markets. With a traditional web-app setup (like a Ruby on Rails project), as soon as you add a new location or update one, the new content is up and live on your website. With a statically generated site, this is not the case. You will need to rebuild the entire site and generate new HTML. The good news is, for a site that doesn't have a ton of content, this static site generation is actually really fast. Like in the tens of seconds fast. A couple of minutes if you consider the entire proecss of minifying assets, building and deploying your site.
+
+We played with and evaluated quite a few options, like pure jekyll, yeoman, Middleman, etc. The one we decided to go with is a very well thought out static site generator called [**Roots**](http://roots.cx/), developed by the fine folks at the Brooklyn based agency, [Carrot Creative](https://carrot.is/). Besides the many nice features that Roots gives you right out of the box (folder structure, asset management and minifcation, great extension and plugin system, etc), they also have a really nifty extension called [**roots-records**](https://github.com/carrot/roots-records). Like most well built things, roots-records serves one purpose and does an amazing job at it. It allows you to hit ANY endpoint that returns a JSON collection and use that collection in your templates to iterate over or if you pass in a template, it will also generate individual static HTML pages for each item in the collection. For example, here is all we had to define in our `app.coffee` file to hit our API endpoing for our market/location:
+
+```coffeescript
+extensions: [
+  records(
+    marketsByCountry: {
+      url:
+        path: "#{process.env.DUBS_API}/api/v1/locations/markets_by_country"
+        headers: { Authorization: "Token token=#{process.env.DUBS_API_TOKEN}" }
+    }
+  )
+]
+```
+
+And here is how this data is referenced inside of our `jade` templates:
+
+```jade
+each countries, index in records.marketsByCountry
+  .row
+    .col-12.col
+      h4
+        != countries[0]
+```
+
+Once we are ready to move our individual location pages over to *static*, all we need to do is pass a `template` option to the `records` block in our app.coffee config, and Roots will generate the individual location pages when it builds the site. This would look something like this:
+
+```coffeescript
+extensions: [
+  records(
+    marketsByCountry: {
+      url:
+        path: "#{process.env.DUBS_API}/api/v1/locations/markets_by_country"
```

Comment:
should we just put an example path like `#{process.env.API_URL_THAT_RETURNS_MARKETS}` and remove all the header stuff rather than show them the actual implementation?